shell/platform/darwin/common/framework/Source/FlutterStandardCodecHelper.cc

```diff
@@ -221,7 +221,7 @@ void FlutterStandardCodecHelperWriteUTF8(CFMutableDataRef data,
     // UTF16 length times 3 will fit all UTF8.
     CFIndex buffer_length = length * 3;
     std::vector<UInt8> buffer;
-    buffer.reserve(buffer_length);
+    buffer.resize(buffer_length);
```

Comment:
Oh wow; nice catch. I'm somewhat surprised this hasn't caused issues other than tipping off ASAN!